client/pages/AssessmentSection/AssessmentSection.tsx

```diff
@@ -1,39 +1,65 @@
 import './AssessmentSection.scss'
-import React from 'react'
+import React, { useEffect, useRef } from 'react'
 import { useTranslation } from 'react-i18next'
 import { useParams } from 'react-router-dom'
 
 import { AssessmentName } from '@meta/assessment'
 
-import { useAssessmentSection } from '@client/store/assessment'
-import { useTableSections } from '@client/store/pages/assessmentSection'
+import { useAppDispatch } from '@client/store'
+import { useAssessment, useAssessmentSection, useCycle } from '@client/store/assessment'
+import { AssessmentSectionActions, useTableSections } from '@client/store/pages/assessmentSection'
 import { useCanEditSection } from '@client/store/user'
+import { useCountryIso } from '@client/hooks'
+import { useIsPrint } from '@client/hooks/useIsPath'
 
 import DataTable from './DataTable'
 import Descriptions, { GeneralComments } from './Descriptions'
 import SectionHeader from './SectionHeader'
 import Title from './Title'
 
-const AssessmentSection: React.FC = () => {
+type Props = {
+  section?: string
+}
+
+const AssessmentSection: React.FC<Props> = (props: Props) => {
+  const { section: sectionProp } = props
+  const assessment = useAssessment()
+  const cycle = useCycle()
+  const countryIso = useCountryIso()
+  const dispatch = useAppDispatch()
   const { i18n } = useTranslation()
   const { assessmentName } = useParams<{ assessmentName: AssessmentName; cycleName: string; section: string }>()
-  const assessmentSection = useAssessmentSection()
-  const tableSections = useTableSections({ sectionName: assessmentSection?.props?.name })
-  const canEditSection = useCanEditSection()
-  const [printView, printOnlyTablesView] = [false, false] // TODO: usePrintView()
+  const assessmentSection = useAssessmentSection(sectionProp)
+  const tableSections = useTableSections({ sectionName: assessmentSection.props?.name })
+  const canEditSection = useCanEditSection(sectionProp)
+  const { print, onlyTables } = useIsPrint()
 
   const panEuropean = assessmentName === AssessmentName.panEuropean
   const disabled = panEuropean || !canEditSection
 
-  if (!assessmentSection) return null
+  const shouldFetchTableSections = useRef(tableSections.length === 0)
+
+  useEffect(() => {
+    if (shouldFetchTableSections.current) {
+      shouldFetchTableSections.current = false
```

Comment:
Why fetching here ?

Reply:
We have section info only here (todo: pass array of sections for endpoint)

Reply:
Don't we do that in the section wrapper ?
We should do it only once